SwiftUIBasics/Views/SignUpView.swift

```diff
@@ -61,12 +111,22 @@ class SignUpViewModel: ObservableObject {
             .assign(to: \.isValidPasswordMatch, on: self)
             .store(in: &cancelableSet)
 
-        Publishers.CombineLatest4($isValidUsernameLength, $isValidPasswordLength, $isValidPasswordUpperCase, $isValidPasswordMatch)
+        Publishers.CombineLatest4($isValidPasswordLowerCase, $isValidPassWordOneSymbol, $isValidPasswordOneNumber, $isValidPre)
+            .map { (a, b, c, d) in
+                return a && b && c && d
+            }
+            .assign(to: \.isValidPre, on: self)
```

Comment:
Circular reference‼️ you are evaluating the binding isValidPre and asigning to the same binding isValidPre